crates/task-impls/src/network.rs

```diff
@@ -182,8 +187,31 @@ impl<TYPES: NodeType> NetworkMessageTaskState<TYPES> {
                     )
                     .await;
                 }
-                DataMessage::DataResponse(_) | DataMessage::RequestData(_) => {
-                    warn!("Request and Response messages should not be received in the NetworkMessage task");
+                DataMessage::DataResponse(response) => {
+                    if let ResponseMessage::Found(message) = response {
+                        match message {
+                            SequencingMessage::Da(da_message) => {
+                                if let DaConsensusMessage::VidDisperseMsg(proposal) = da_message {
+                                    broadcast_event(
+                                        Arc::new(HotShotEvent::VidResponseRecv(sender, proposal)),
+                                        &self.internal_event_stream,
+                                    )
+                                    .await;
+                                }
+                            }
+                            SequencingMessage::General(_) => {}
```

Comment:
Should this also include responses for proposal? Or is that handled by a different message type?

Reply:
we have those `GeneralConsensusMessage`. Im not sure how easy it would be to refactor them into here. I need to think a little more on this

Reply:
If we want to move them we can probably make a separate task